contracts/verifiers/ZKPVerifierBase.sol

```diff
@@ -34,6 +49,21 @@ abstract contract ZKPVerifierBase is IZKPVerifier, ContextUpgradeable {
         }
     }
 
+    function _setState(IState state) internal {
+        _getZKPVerifierStorage()._state = state;
+    }
+
+    using VerifierLib for ZKPVerifierStorage;
+
+    function __ZKPVerifierBase_init(IState stateCrossChain) internal onlyInitializing {
+        __ZKPVerifierBase_init_unchained(stateCrossChain);
+    }
+
+    function __ZKPVerifierBase_init_unchained(IState state) internal onlyInitializing {
+        ZKPVerifierStorage storage $ = _getZKPVerifierStorage();
+        $._state = state;
+    }
```

Comment:
Maybe this could be simplified with this

```suggestion
    function __ZKPVerifierBase_init(IState stateCrossChain) internal onlyInitializing {
        _setState(stateCrossChain);
    }
```

Reply:
It's better to keep `__ZKPVerifierBase_init` as it is and change only `__ZKPVerifierBase_init_unchained` as it's kind of a conventions set by OpenZeppelin to have two initialisers, one, which call parent intitalizers (_init) and the other one which does not call them (_init_unchained)